doc/source/release/1.15.0-notes.rst

```diff
@@ -17,96 +17,1431 @@ run your code with ``python -Wd`` and check for ``DeprecationWarning`` s).
+- `scipy.stats.zmap`, `scipy.stats.zscore`, and `scipy.stats.gzscore`
+- `scipy.stats.tmean`, `scipy.stats.tvar`, `scipy.stats.tstd`,
+  `scipy.stats.tsem`, `scipy.stats.tmin`, and `scipy.stats.tmax`
+- `scipy.stats.gmean`, `scipy.stats.hmean` and `scipy.stats.pmean`
+- `scipy.stats.combine_pvalues`
+- `scipy.stats.ttest_ind`, `scipy.stats.ttest_rel`
+- `scipy.stats.directional_stats`
+- `scipy.ndimage` functions will now delegate to ``cupyx.scipy.ndimage``,
+  and for other backends will transit via NumPy arrays on the host.
+
+
+*******************
+Deprecated features
+*******************
+- Functions `scipy.linalg.interpolative.rand` and
+  `scipy.linalg.interpolative.seed` have been deprecated and will be removed
+  in SciPy ``1.17.0``.
+- Complex inputs to `scipy.spatial.distance.cosine` and
+  `scipy.spatial.distance.correlation` have been deprecated and will raise
+  an error in SciPy ``1.17.0``.
+- `scipy.stats.find_repeats` is deprecated as of SciPy ``1.15.0`` and will be
+  removed in SciPy ``1.17.0``. Please use
+  ``numpy.unique``/``numpy.unique_counts`` instead.
+- `scipy.linalg.kron` is deprecated in favour of ``numpy.kron``.
+- Using object arrays and longdouble arrays in `scipy.signal`
+  convolution/correlation functions (`scipy.signal.correlate`,
+  `scipy.signal.convolve` and `scipy.signal.choose_conv_method`) and
+  filtering functions (`scipy.signal.lfilter`, `scipy.signal.sosfilt`) has
+  been deprecated as of SciPy ``1.15.0`` and will be removed in SciPy
+  ``1.17.0``.
+- `scipy.stats.linregress` has deprecated one-argument use; the two
+  variables must be specified as separate arguments.
+- ``scipy.stats.trapz`` is deprecated in favor of `scipy.stats.trapezoid`.
+- `scipy.special.lpn` is deprecated in favor of `scipy.special.legendre_p_all`.
+- `scipy.special.lpmn` and `scipy.special.clpmn` are deprecated in favor of
+  `scipy.special.assoc_legendre_p_all`.
+
+********************
+Expired Deprecations
+********************
+- The wavelet functions in `scipy.signal` have been removed. This includes
+  ``daub``, ``qmf``, ``cascade``, ``morlet``, ``morlet2``, ``ricker``,
+  and ``cwt``. Users should use ``pywavelets`` instead.
+- ``scipy.signal.cmplx_sort`` has been removed.
+- ``scipy.integrate.quadrature`` and ``scipy.integrate.romberg`` have been
+  removed in favour of `scipy.integrate.quad`.
+- ``scipy.stats.rvs_ratio_uniforms`` has been removed in favor of
+  `scipy.stats.sampling.RatioUniforms`.
+- `scipy.special.factorial` now raises an error for non-integer scalars when
+  ``exact=True``.
+- `scipy.integrate.cumulative_trapezoid` now raises an error for values of
+  ``initial`` other than ``0`` and ``None``.
+- Complex dtypes now raise an error in `scipy.interpolate.Akima1DInterpolator`
+  and `scipy.interpolate.PchipInterpolator`
+- ``special.btdtr`` and ``special.btdtri`` have been removed.
+- The default of the ``exact=`` kwarg in ``special.factorialk`` has changed
+  from ``True`` to ``False``.
 
```

Comment:
```suggestion
- All functions in the ``scipy.misc`` submodule have been removed.

```

Reply:
so not the module itself 🤔 ?

Reply:
IMHO I think a fail-fast approach would've been better; this way it'll inevitably result in a delayed `AttributeError` (unless the import is redundant), as opposed to an `ImportError` from the crime scene itself. 
Empty modules also can't be marked with `@decorated` like functions and classes can, so type-checkers won't be helpful as witnesses.

But what's done is done I suppose 🤷🏻

Reply:
This is still being discussed to some extent at https://github.com/scipy/scipy/issues/15765#issuecomment-2519263147.